hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -461,7 +465,7 @@ public void refreshTimeline() throws IOException {
     if (!hasErrors || cfg.commitOnErrors) {
       HashMap<String, String> checkpointCommitMetadata = new HashMap<>();
       checkpointCommitMetadata.put(CHECKPOINT_KEY, checkpointStr);
-      if (cfg.checkpoint != null) {
+      if (cfg.checkpoint != null && !"timestamp".equals(props.getString("hoodie.deltastreamer.source.kafka.checkpoint.type"))) {
```

Comment:
Can you help me understand why we need this ? My understanding is that, user will set cfg.checkpoint during first batch and set the checkpoint type (to timestamp) as well. but even for any checkpoint types, we should add the checkpoint_reset_key here at the end of 1st batch. Am I missing something. can you please help me understand.

Reply:
You understand that is correct, I wanted to set the timestamp to CHECKPOINT_RESET_KEY at the time. Considering that it could not serve a practical purpose, I cancelled it. After listening to your thoughts, adding it should be more appropriate